File: src/models/hotel.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Hotel extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      Hotel.belongsTo(models.User,{
        foreignKey: "user_id",
        targetKey: "id",
        as: "userData",
      })
      Hotel.belongsTo(models.Province,{
        foreignKey: "province_id",
        targetKey: "id",
        as: "provinceData",
      })
      Hotel.hasMany(models.HotelRoom,{
        foreignKey: "hotel_id",
        as: "hotelromData",
      })
      Hotel.hasMany(models.Booking ,{
        foreignKey: "hotel_id",
        as: "hotelbookingData",
      })
    }
  }
  Hotel.init({
    user_id : DataTypes.INTEGER, 
    province_id  : DataTypes.INTEGER,
    hotel_name : DataTypes.STRING,
    hotel_desc : DataTypes.TEXT,
    hotel_address : DataTypes.STRING,
    hotel_phone : DataTypes.STRING,
    hotel_email : DataTypes.STRING,
    room_quantity : DataTypes.INTEGER,
    image : DataTypes.STRING,
  }, {
    sequelize,
    modelName: 'Hotel',
  });
  return Hotel;
};

File: Databases/Entities/Hotel.cs




using System.ComponentModel.DataAnnotations;
using HotelApp.API.Database.Entities;

namespace HotelApp.API.Databases.Entities
{
    public class Hotel
    {
        [Key]
        public int Id { get; set; }

        [StringLength(100)]
        public string Hotel_name { get; set; } = null!;

        public int Room_quantity { get; set; }


        public string Hotel_description { get; set; } = null!;
        [StringLength(255)]
        public string Hotel_address { get; set; } = null!;
        [StringLength(255)]
        public string Hotel_phone { get; set; } = null!;

        [StringLength(255)]
        public string Hotel_email { get; set; } = null!;
        [StringLength(255)]
        public string? Image { get; set; } = null!;

        public int ProvinceId { get; set; }
        public virtual Province Province { get; set; } = null!;

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public virtual ICollection<HotelRoom> HotelRooms { get; set; } = new HashSet<HotelRoom>();
        public virtual ICollection<Booking> Bookings { get; set; } = new HashSet<Booking>();
    }
}